python/paddle/fluid/tests/unittests/test_fleet_sharding_meta_optimizer.py

```diff
@@ -293,5 +292,200 @@ def test_sharding_clone_for_test(self):
         ])
 
 
+class TestFleetMetaOptimizer(TestFleetMetaOptimizer):
+    def setUp(self):
+        os.environ["PADDLE_TRAINER_ID"] = "3"
```

Comment:
666

Reply:
in  mp-sharding or  sharding-hybrid-dp setting, we need at least 4 workers to setting up the parallelism logic.